src/applications/hca/tests/config/serviceInformation.unit.spec.js

```diff
@@ -1,88 +1,89 @@
 import React from 'react';
 import { expect } from 'chai';
 import sinon from 'sinon';
-import ReactTestUtils from 'react-dom/test-utils';
+import { mount } from 'enzyme';
 import moment from 'moment';
 
-import { DefinitionTester, getFormDOM } from '../../../../platform/testing/unit/schemaform-utils.jsx';
+import { DefinitionTester, fillData, fillDate } from '../../../../platform/testing/unit/schemaform-utils.jsx';
 import formConfig from '../../config/form';
 
 describe('Hca serviceInformation', () => {
   const { schema, uiSchema } = formConfig.chapters.militaryService.pages.serviceInformation;
-  const definitions = formConfig.defaultDefinitions;
-  it('should render', () => {
-    const form = ReactTestUtils.renderIntoDocument(
+
+  it('renders military info', () => {
+    const form = mount(
       <DefinitionTester
+        definitions={formConfig.defaultDefinitions}
         schema={schema}
         data={{}}
-        uiSchema={uiSchema}
-        definitions={definitions}/>
+        formData={{}}
+        uiSchema={uiSchema}/>
     );
-    const formDOM = getFormDOM(form);
 
-    expect(formDOM.querySelectorAll('input').length).to.equal(2);
-    expect(formDOM.querySelectorAll('select').length).to.equal(6);
+    expect(form.find('input').length).to.equal(2);
+    expect(form.find('select').length).to.equal(5);
   });
 
-  it('should not submit empty form', () => {
+  it('does not submit without info', () => {
     const onSubmit = sinon.spy();
-    const form = ReactTestUtils.renderIntoDocument(
+    const form = mount(
       <DefinitionTester
+        definitions={formConfig.defaultDefinitions}
         schema={schema}
-        definitions={definitions}
+        data={{}}
+        formData={{}}
         onSubmit={onSubmit}
         uiSchema={uiSchema}/>
     );
 
-    const formDOM = getFormDOM(form);
+    form.find('form').simulate('submit');
+    expect(form.find('.usa-input-error').length).to.equal(3);
 
-    formDOM.submitForm();
-
-    expect(formDOM.querySelectorAll('.usa-input-error').length).to.equal(4);
```

Comment:
We no longer require discharge type by default, and require it when discharge date is in the present or past.